### PR TITLE
feat: Render guild territories on main map

### PR DESCRIPTION
**Pull request by kristofbolyai**
image

### CODE REVIEW

File: common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java [hunk 1]
```diff
@@ -340,7 +345,7 @@ public boolean mouseClicked(double mouseX, double mouseY, int button) {
                 return true;
             }
 
-            if (hovered != null) {
+            if (hovered != null && !(hovered instanceof TerritoryPoi)) {
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java:348]: Maybe move to onClick code to the hovered poi itself?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java:348]: We could eventually. Didn't want to refactor Pois even more.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java:348]: I dont see that many use cases though

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java:348]: Seaskipper map will have custom behavior on hover click, but yeah.

File: common/src/main/java/com/wynntils/core/webapi/profiles/TerritoryProfile.java [hunk 2]
```diff
@@ -164,16 +167,21 @@ public TerritoryProfile deserialize(JsonElement json, Type typeOfT, JsonDeserial
             }
 
             String guildPrefix;
-            if (territory.get("guildPrefix").isJsonNull()) guildPrefix = "UNK";
-            else guildPrefix = territory.get("guildPrefix").getAsString();
+            if (territory.get("guildPrefix").isJsonNull()) {
+                guildPrefix = "UNK";
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/core/webapi/profiles/TerritoryProfile.java]: Could this be an actual guild prefix? Maybe a prefix of UNKWN or something would be better

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/webapi/profiles/TerritoryProfile.java]: I know it differs from legacy

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/webapi/profiles/TerritoryProfile.java]: Fixed, shouldnt happen though.

**Review by magicus — APPROVED** [review]

**Comment by magicus** [comment]
The use of <kbd>control</kbd> is lacking a bit in discoverability. :-) Maybe we'll need some kind of legend/help text for the map soon. Maybe a `?` icon that when hovered over shows a tooltip with help..? That can be both non-intrusive but also clearly accessible for new users. (Not requesting this as part of this PR, just throwing ideas randomly around me)

**Comment by kristofbolyai** [comment]
> The use of <kbd>control</kbd> is lacking a bit in discoverability. :-) Maybe we'll need some kind of legend/help text for the map soon. Maybe a `?` icon that when hovered over shows a tooltip with help..? That can be both non-intrusive but also clearly accessible for new users. (Not requesting this as part of this PR, just throwing ideas randomly around me)

Yeah, I just ported legacy stuff. We can work on this when we add gui for map.

**Pull request merged**